court/tests.py

```diff
@@ -28,15 +28,16 @@ def test_create_valid_court(self):
                              city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30).full_clean()
 
     def test_create_three_courts_without_id(self):
-        court_id1 = Court.objects.create(x=Decimal('11'), y=Decimal('22'),
-                                         city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
-        court_id2 = Court.objects.create(x=Decimal('22'), y=Decimal('33'),
-                                         city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=40)
-        court_id3 = Court.objects.create(x=Decimal('33'), y=Decimal('44'),
-                                         city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=50)
-        assert court_id1.courtID == 1
-        assert court_id2.courtID == 2
-        assert court_id3.courtID == 3
+        court1 = Court.objects.create(x=Decimal('11'), y=Decimal('22'),
+                                      city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+        court2 = Court.objects.create(x=Decimal('22'), y=Decimal('33'),
+                                      city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=40)
+        court3 = Court.objects.create(x=Decimal('33'), y=Decimal('44'),
+                                      city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=50)
+        first_id = court1.courtID
+        assert court1.courtID == first_id
+        assert court2.courtID == first_id+1
+        assert court3.courtID == first_id+2
```

Comment:
You find yourself having to change this test in order to verify that court ID is as expected, but I don't think you actually have any obligations or trying to control the logic according to which IDs are being assigned.

If we're not trying to control the the IDs then what are we trying to test here? Are we trying to test that we can create courts without ID? Are trying to verify that they are getting new IDs? That there are actually new courts?

I suggest that you establish what is it that you're trying to test and then test that rather than testing the logic for assigning court IDs which I don't think we control nor we rely on. Suppose tomorrow you upgrade django and the new version defaults to assigning IDs randomly instead of with running numbers. You'll have regressions on this test that you had no reason to have.

Reply:
This test was in the last demo, but with different variable names.
The problem that I had encountered is that I couldn't figure how to set the ID of the court, without assigning one.
I solved it by writing the function `def save(self, *args, **kwargs)` in `court/models.py`, which generates an ID for a court if none is assigned.
So in this test, what I'm testing is that if no ID is assigned, we'll get a generated one. So if 3 courts are made one after the other, they should have ascending order of IDs (firstID, firstID+1, firstID+2).

I agree that in some edge cases the test would fail (what if a new court is created between court1 is made, and before court2 is made? court2 will get firstID+2 instead of firstID+1). But if I remember correctly, in the previous PR (create court model) we agreed that this is a better implementation for the test, rather the older one.

Reply:
Alright. I didn't have that logic in mind. In that case, the test makes sense.